ably/ably-chat-js
# demo-app: add occupancy to the demo app.

### Context

* [CHA-345]
*  The demo app did not use the occupancy capability of the chat SDK, so this PR adds it's usage.

### Description

* Added Occupancy component to render the occupancy metric on screen.  
* Added occupancy hook to receive occupancy events and update the metric.
* Added some basic styling.

<img width="765" alt="Screenshot 2024-07-17 at 19 59 09" src="https://github.com/user-attachments/assets/e81bc1a1-8f15-4b55-a195-e1e16bd9ff16">


### Checklist


### Testing Instructions (Optional)

* Spin up the demo app and open it in a few tabs, you should see the occupancy metric update accordingly.


[CHA-345]: https://ably.atlassian.net/browse/CHA-345?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
